### DISCUSSION TITLE
Check ideas

### DISCUSSION THREAD

**Discussion by flo-at** [flo-at/minmon#31] [Ideas]
- Filesystem inode usage
- Filesystem free space (in addition to the usage check)
- Folder size
- S.M.A.R.T.
- Ping
- HTTP response, keyword, ..
- HTTPS certificate expiration date
- Podman container status
- Process (integer) output
- [OpenMetrics](https://github.com/OpenObservability/OpenMetrics)

**Comment by chenxiaolong** [comment]
Would it be possible to extend the `ProcessExitStatus` check to capture the output of the command and make that available as a placeholder? (Maybe truncated to ensure a check can't cause minmon to allocate excessive amounts of memory?) One of my process checks, for example, looks for ZFS replication bookmarks that aren't recent, and then prints out a summary. It'd be nice if that could be made visible in the alerts.

By the way, thanks for creating minmon! It's exactly what I was looking for: a way to run commands periodically and email me N repeats after a state transition. I have no need for web UIs, graphs, data collection for things I didn't explicitly configure, etc. Minmon is simple, yet very flexible thanks to the process checks and actions. I also very much appreciate that it doesn't fundamentally require root to run.

**Reply by flo-at** [reply]
Glad you like it! I'll have some spare time the next few days. I'll take a look at it, should be an easy change. Maybe I'll also try to get the filter branch ready for merging while I'm on it.
Cheers!

**Reply by flo-at** [reply]
[There you go!](https://github.com/flo-at/minmon/releases/tag/v0.6.0)

**Reply by chenxiaolong** [reply]
Works perfectly, thanks!

**Comment by seang96** [comment]
Https would be cool for its usability. One not suggested is prometheus checks. It would be nice as a separate check for easier configuration.

**Reply by flo-at** [reply]
Could you give an example or some details? By https you mean certificate expiration checks? And Prometheus to store the check results?

Https cert validity has been suggested before, it's a good idea for sure and shouldn't be too hard to implement.

**Reply by seang96** [reply]
Sorry I was meaning making web api calls and checking the response code / body.

For prometheus I mean monitor a metric and trigger an action after the configured threshold is reached.

**Reply by flo-at** [reply]
The code for the http(s) part should be mostly identical to the webhook action.

I don't know too much about prometheus but isn't one of it's main features to act when it's time series data hits a threshold?
I'll take a look at the API soon to see how much effort that would be.

**Reply by seang96** [reply]
I actually don't have it installed so I'm not sure. I do have a lot of services that support the metrics api though and I imagine just alerts from minmon would be lighter on resources than hosting prometheus.

**Reply by flo-at** [reply]
Okay, I think I get it now. You'd like to see MinMon "act" as Prometheus so other services can send their metrics and MinMon uses them for the checks. Right? That's a pretty cool idea to be ale to monitor everything that already implements the Prometheus API with a single check.

Edit: There seems to be an initiative to make this into a standard called [OpenMetrics](https://github.com/OpenObservability/OpenMetrics)
Edit2: Prometheus actively pulls the metrics (which makes sense for monitoring), I mixed that up above.
